data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/file/FileWatcher.java

```diff
@@ -136,25 +140,36 @@ private void update() throws IOException {
     try (
       final var fileReader = new FileReader(toWatch);
       final var bufferedReader = new BufferedReader(fileReader)) {
-      parseFromJson(bufferedReader);
+      final var contract = parseFromJson(bufferedReader);
+      final var previousLastContract = this.lastContract;
+      this.lastContract = contract;
+      if (contract == null || contract.equals(previousLastContract)) {
```

Comment:
More than doing `contract.equals`, can you check just that the generation number is equal?